softgroup/basis/common-token/src/main/java/com/softgroup/token/impl/TokenService.java

```diff
@@ -83,55 +87,32 @@ public boolean validateToken(String token,TokenType type) {
     @Override
     public JwtUserIdentifier getUserIdentifier(String token, TokenType type) {
         //Parse and validate token
+        try {
+            if(token==null){
+                throw new IllegalArgumentException("Token cannot be null. ");
+            }
 
-        if(token==null){
-            return null;
-        }
-
-        Jws<Claims> jws = parseTokenBody(token,keyFactory.getKey(type));
-
-        if(jws==null){
-            return null;
-        }
-
-        //Validate expiration date
-        if(jws.getBody().getExpiration().before(new Date())){
-            return null;
-        }
+            Jws<Claims> jws = parseTokenBody(token,keyFactory.getKey(type));
 
-        if(jws.getBody().getIssuedAt().after(new Date())){
-            return null;
-        }
+            if(jws.getBody().getIssuedAt().after(new Date())){
+                throw new JwtException("Error.This token created in future");
+            }
 
-        //parse type of token
+            return new JwtUserIdentifier(jws.getBody().get("userID",String.class),jws.getBody().get("deviceID",String.class));
 
-        if (!jws.getBody().get("tokenType").equals(type.name())){
+        } catch (JwtException exception) {
+            //log this jwt error
+            exception.printStackTrace();
```

Comment:
@Kun-Kun If you can deal with exception here - than don't throw it. If you can not deal with it - don't catch it ))